src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -875,13 +887,18 @@ private bool IsAttributeDeclarationTerminator()
                 || this.IsPossibleAttributeDeclaration(); // start of a new one...
         }
 
-        private AttributeListSyntax ParseAttributeDeclaration()
+        private AttributeListSyntax? TryParseAttributeDeclaration(bool inExpressionContext)
         {
-            if (this.IsIncrementalAndFactoryContextMatches && this.CurrentNodeKind == SyntaxKind.AttributeList)
+            if (this.IsIncrementalAndFactoryContextMatches &&
+                this.CurrentNodeKind == SyntaxKind.AttributeList &&
+                !inExpressionContext)
```

Comment:
Do we have incremental parsing tests covering this change?

Reply:
no.  but i'll add

Reply:
added.